SwanCustomEnvironments/swancustomenvironments/scripts/makenv.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+
+# Author: Rodrigo Sobral 2024
+# Copyright CERN
+# This script allows to create a venv virtual environment to use in notebooks and terminals. The environment contains the packages from a provided repository.
+
+_log () {
+    if [ "$*" == "ERROR:"* ] || [ "$*" == "WARNING:"* ] || [ "${JUPYTER_DOCKER_STACKS_QUIET}" == "" ]; then
+        echo "$@"
+    fi
+}
+
+# Function to ensure a unique folder name, instead of overwriting an existing one
+define_repo_path() {
+    local folder_name=$1
+    local counter=0
+    
+    local INIT_REPO_PATH="$HOME/SWAN_projects/${folder_name}"
+    REPO_PATH="${INIT_REPO_PATH}"
+
+    # Loop to find a unique folder name
+    while [ -d "$REPO_PATH" ]; do
+        counter=$((counter + 1))
+        REPO_PATH="${INIT_REPO_PATH}_${counter}"
+    done
+
+    # Warn the backend of the new folder name
+    if [ $counter -ne 0 ]; then
+        echo "FOLDER_NAME_CHANGE:${folder_name}_${counter}"
+    fi
+}
+
+# Function for printing the help page
+print_help() {
+    _log "Usage: makenv --env/-e NAME --repo/-r REPOSITORY [--accpy ACCPY_VERSION] [--help/-h]"
+    _log "Options:"
+    _log "  -e, --env NAME              Name of the custom virtual environment (mandatory)"
+    _log "  -r, --repo REPOSITORY       Path or http link for a public repository (mandatory)"
+    _log "  -h, --help                  Print this help page"
+    _log "  --accpy VERSION             Version of Acc-Py to be used"
+}
+
+# --------------------------------------------------------------------------------------------
```

Comment:
Usually if you do this sort of separation, which is OK, you should include some header comment about that the section below is about.